sdk/bento/example/mountcar.py

```diff
@@ -0,0 +1,103 @@
+#
+# bentobox-sdk
+# mountain car example simulation
+#
+
+from bento import types
+from bento.graph.plotter import Plotter
+from bento.spec.ecs import ComponentDef, EntityDef
+
+from bento.example.specs import Velocity, Position
+from bento.sim import Simulation
+from bento.spec.sim import SimulationDef
+
+Action = ComponentDef(
+    name="action",
+    schema={
+        "accelerate": types.int32,
+    },
+)
+"""
+Action allows the agent to control the car in the Simulation via its acceleration.
+
+Attributes:
+    **accelerate** (`types.int32`): Set the acceleration of the car.
+
+    - 0: Accelerate to the Left
+    - 1: Don't accelerate
+    - 2: Accelerate to the Right
+"""
+
+State = ComponentDef(
+    name="state",
+    schema={
+        "reward": types.int32,
+        "ended": types.boolean,
+    },
+)
+"""
+State tracks the current state of the Simulation.
+
+Attributes:
+    **reward** (`types.int32`): Reward given to agent for the current Simulation step:
+
+    - Reward of 0 is awarded if the agent reached the flag (`Position.x >= 0.5`) on top of the mountain.
+    - Reward of -1 is penalized if the position of the agent `Position.x < 0.5`
+
+    **ended** (`types.boolean`): Whether the Simulation is has ended (`Position.x > 0.5`).
+"""
+
+MountainCar = SimulationDef(
+    name="mountain_car",
+    components=[Velocity, Position, Action, State],
+    entities=[
+        EntityDef(components=[Velocity, Position]),
+        EntityDef(components=[Action, State]),
+    ],
+)
+"""
+Mountain Car Simulation implemented using `bento`
+
+A car is started at random position the bottom of a valley.
+The agent may choose to accelerate the car to the left, right or cease any acceleration.
+The objective of the Simulation is to reach the flag on top of the mountain at `Position.x > 0.5`
+The simulation ends when the car's `Position.x > 0.5`
+"""
+
+
+@MountainCar.init
+def init_fn(g: Plotter):
+    car = g.entity(components=[Velocity, Position])
+    car[Velocity].x = 0.0
+    car[Position].x = g.random(-0.6, -0.4)
+
+    env = g.entity(components=[Action, State])
+    env[State].reward = 0
+    env[State].ended = False
+    env[Action].accelerate = 1
+
+
+@MountainCar.system
+def sim_fn(g: Plotter):
+    env = g.entity(components=[Action, State])
+    car = g.entity(components=[Velocity, Position])
+
+    # process car physics
+    # compute velocity based on acceleration action & decceleration due to gravity
+    acceleration, gravity, max_speed = 0.001, 0.0025, 0.07
```

Comment:
Would be great to have these constants defined in terms of some sort of time value. For example, acceleration may be 5 m/s^2, and we define a time step to be 0.1s, so we can calculate that the acceleration per step is approximately 0.5 m/s

Reply:
These constants where lifted off the original [OpenAI Implmentation](https://github.com/openai/gym/blob/master/gym/envs/classic_control/mountain_car.py). 
- The original  reference implementation is also not  clear on what the real life equivalent units of these constants are.
- Since we don't car about real time (ie no graphics), i think its safe to assume a constant time per step, so there is no need to calculate based on time.